machines/issuersMachine.ts

```diff
@@ -232,19 +315,36 @@ export const IssuersMachine = model.createMachine(
     actions: {
       setIssuers: model.assign({
         issuers: (_, event) => event.data,
+      }),
+      setNoInternet: model.assign({
+        errorMessage: 'noInternetConnection',
+      }),
+      setDownloadingCreds: model.assign({
+        loadingReason: 'downloadingCredentials',
+      }),
+      setTokenLoadingReason: model.assign({
+        loadingReason: 'settingUp',
+      }),
+      unsetLoadingReason: model.assign({
         loadingReason: null,
       }),
-
       setError: model.assign({
         errorMessage: (_, event) => {
-          console.log('Error while fetching issuers ', event.data.message);
-          return event.data.message === 'Network request failed'
-            ? 'noInternetConnection'
-            : 'generic';
+          console.log('Error occured ', event.data.message);
+          const error = event.data.message;
+          switch (error) {
+            case 'Network request failed':
```

Comment:
can we add 'Network request failed' and 'request timedout' errors as well in the error enum